[BUG] npm ci fails with ERR_FS_EISDIR for node_modules/.bin

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Can't npm ci if there's an existing .bin directory in node_modules

### Expected Behavior

It deletes node_modules without crashing

### Steps To Reproduce

Use this package.json

```json
{
  "name": "npm-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "eslint": "^8.31.0"
  }
}
```

* `npm ci`
* `npm ci`
* It fails the second time
```
npm ERR! code ERR_FS_EISDIR
npm ERR! syscall rm
npm ERR! path /Users/justintoman/code/npm-test/node_modules/.bin
npm ERR! errno 21
npm ERR! Path is a directory: rm returned EISDIR (is a directory) /Users/justintoman/code/npm-test/node_modules/.bin

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/justintoman/.npm/_logs/2023-01-12T21_14_19_211Z-debug-0.log
```

### Environment

- npm: 9.3.0
- Node.js: 18.12.1
- OS Name: MacOS Venture 13.1
- System Model Name: Macbook Pro
- npm config:
```ini
; node bin location = /Users/justintoman/.nvm/versions/node/v18.12.1/bin/node
; node version = v18.12.1
; npm local prefix = /Users/justintoman/code/npm-test
; npm version = 9.3.0
; cwd = /Users/justintoman/code/npm-test
; HOME = /Users/justintoman
; Run `npm config ls -l` to show all defaults.
```

--- a/lib/commands/ci.js
+++ b/lib/commands/ci.js
@@ -67,7 +67,7 @@ class CI extends ArboristWorkspaceCmd {
       const path = `${where}/node_modules`
       // get the list of entries so we can skip the glob for performance
       const entries = await fs.readdir(path, null).catch(er => [])
-      return Promise.all(entries.map(f => fs.rm(`${path}/${f}`, { force: true })))
+      return Promise.all(entries.map(f => fs.rm(`${path}/${f}`, { force: true, recursive: true })))
     })
 
     await arb.reify(opts)

--- a/test/lib/commands/ci.js
+++ b/test/lib/commands/ci.js
@@ -79,6 +79,30 @@ t.test('reifies, audits, removes node_modules', async t => {
   t.equal(fs.existsSync(nmAbbrev), true, 'installs abbrev')
 })
 
+t.test('reifies, audits, removes node_modules on repeat run', async t => {
+  const { npm, joinedOutput, registry } = await loadMockNpm(t, {
+    prefixDir: {
+      abbrev: abbrev,
+      'package.json': JSON.stringify(packageJson),
+      'package-lock.json': JSON.stringify(packageLock),
+      node_modules: { test: 'test file that will be removed' },
+    },
+  })
+  const manifest = registry.manifest({ name: 'abbrev' })
+  await registry.tarball({
+    manifest: manifest.versions['1.0.0'],
+    tarball: path.join(npm.prefix, 'abbrev'),
+  })
+  registry.nock.post('/-/npm/v1/security/advisories/bulk').reply(200, {})
+  await npm.exec('ci', [])
+  await npm.exec('ci', [])
+  t.match(joinedOutput(), 'added 1 package, and audited 2 packages in')
+  const nmTest = path.join(npm.prefix, 'node_modules', 'test')
+  t.equal(fs.existsSync(nmTest), false, 'existing node_modules is removed')
+  const nmAbbrev = path.join(npm.prefix, 'node_modules', 'abbrev')
+  t.equal(fs.existsSync(nmAbbrev), true, 'installs abbrev')
+})
+
 t.test('--no-audit and --ignore-scripts', async t => {
   const { npm, joinedOutput, registry } = await loadMockNpm(t, {
     config: {

